File: randomProblems/MonkeyMayhem.py
num_top, num_left = map(int, input().split())
left = [int(n) - m for m, n in enumerate(input().split())]
top = [int(n) for n in input().split()]

collisions = 0

for x, top_monkey in enumerate(top):
	if top_monkey != -1:
		length = len(left)
		left = list(filter(lambda a: a != top_monkey - x, left))
		collisions += length - len(left)

print(collisions)

File: randomProblems/MonkeyMayhem.cpp
#include <iostream>
using namespace std;


void printArray(int arr[], int size) {
	cout << "ARRAY: ";
	for (int i = 0; i < size; i++) {
		cout << arr[i] << " ";
	}
	cout << endl;
}


int main() {
	int num_top;
	int num_left;
	cin >> num_left;
	cin >> num_top;

	int top[num_top];
	int left[num_left];

	for (int i = 0; i < num_left; i++) {
		cin >> left[i];
	}

	for (int i = 0; i < num_top; i++) {
		cin >> top[i];
	}

	//printArray(top, num_top);
	//printArray(left, num_left);


	int collisions = 0;
	for (int x = 0; x < num_top; x++) {
		if (top[x] == -1)
			continue;

		for (int y = 0; y < num_left; y++) {
			if (left[y] == -1)
				continue;

			if (left[y] - y == top[x] - x) {
				collisions++;
				left[y] = -1;
				break;
			}
		}
	}

	cout << collisions << endl;

	return 0;
}